eng/pipelines/templates/jobs/live.tests.yml

```diff
@@ -93,12 +93,22 @@ jobs:
           JdkVersion: $(JavaTestVersion)
 
       - ${{ if ne(parameters.DisableAzureResourceCreation, 'true') }}:
+        - pwsh: |
+            Write-Host "UseFederatedAuth: ${{ parameters.UseFederatedAuth }}"
+          displayName: Use federated auth ${{ parameters.UseFederatedAuth }}
+
+        - ${{ if parameters.UseFederatedAuth }}:
+          - pwsh: |
+              Write-Host "hello world"
+            displayName: Template things UseFederatedAuth is true
+
         - template: /eng/common/TestResources/build-test-resource-config.yml
           parameters:
             SubscriptionConfiguration: ${{ parameters.CloudConfig.SubscriptionConfiguration }}
             SubscriptionConfigurations: ${{ parameters.CloudConfig.SubscriptionConfigurations }}
             EnvVars: ${{ parameters.EnvVars }}
-            SubscriptionConfigurationFilePaths: ${{ parameters.CloudConfig.SubscriptionConfigurationFilePaths }}
+            ${{ if parameters.UseFederatedAuth }}:
+              SubscriptionConfigurationFilePaths: ${{ parameters.CloudConfig.SubscriptionConfigurationFilePaths }}
```

Comment:
How are these connected? What harm is it to add an empty list of paths?

Reply:
In the case of `UseFederatedAuth: false` the default subscription configuration includes a [default sub config file](https://github.com/Azure/azure-sdk-for-java/blob/main/eng/pipelines/templates/stages/archetype-sdk-tests-isolated.yml#L65-L66).

`AzurePublicMsft.json` has empty string values for the test and provisioner app secrets. Having no test secret breaks service principal auth (in use today). 

We could remove those empty values from `AzurePublicMsft.json` and [we should be OK](https://github.com/Azure/azure-sdk-for-java/blob/main/eng/common/TestResources/New-TestResources.ps1#L118-L123) but the sub config DevOps variable would still have the secrets from when configs were merged.

Reply:
I like clearing the secrets out. I just didn't realize folks were getting anything in the paths set yet. Also the error folks were seeing `Cannot bind argument to parameter 'Path' because it is an empty string.` didn't make any since to me based on this issue.

Reply:
This happens because `CloudConfig` goes through several iterations on its way down the templates. Ultimately, if the key isn't present in one of the higher order objects it looks like DevOps is just substituting an empty string in for the value which `convertToJson` turns into a pair of double quotes (`""`).